homeassistant/components/synology_dsm/media_source.py

```diff
@@ -0,0 +1,243 @@
+
+            ret = []
+            ret.append(
+                BrowseMediaSource(
+                    domain=DOMAIN,
+                    identifier=f"{item.identifier}/0",
+                    media_class=MediaClass.DIRECTORY,
+                    media_content_type=MediaClass.IMAGE,
+                    title="All images",
+                    can_play=False,
+                    can_expand=True,
+                )
+            )
+            for album in albums:
+                ret.append(
+                    BrowseMediaSource(
+                        domain=DOMAIN,
+                        identifier=f"{item.identifier}/{album.album_id}",
+                        media_class=MediaClass.DIRECTORY,
+                        media_content_type=MediaClass.IMAGE,
+                        title=album.name,
+                        can_play=False,
+                        can_expand=True,
+                    )
+                )
+
+            return ret
+
+        # Request items of album
+        # Get Items
+        # The library works sync, and this expects async calls
+        album = SynoPhotosAlbum
+        album.album_id = int(identifier_parts[1])
+        try:
+            items = await diskstation.api.photos.get_items_from_album(album, 0, 1000)
+        except SynologyDSMException:
+            return []
+
+        ret = []
+        for items_item in items:
+            mime_type, _ = mimetypes.guess_type(items_item.file_name)
+            assert isinstance(mime_type, str)
+            if mime_type.startswith("image/"):
+                ret.append(
+                    BrowseMediaSource(
+                        domain=DOMAIN,
+                        identifier=f"{identifier_parts[0]}/{identifier_parts[1]}/{items_item.thumbnail_cache_key}/{items_item.file_name}",
+                        media_class=MediaClass.IMAGE,
+                        media_content_type=mime_type,
+                        title=items_item.file_name,
+                        can_play=True,
+                        can_expand=False,
+                        thumbnail=await self.async_get_thumbnail(
+                            items_item.thumbnail_cache_key,
+                            items_item.item_id,
+                            "sm",
+                            identifier_parts[0],
+                        ),
+                    )
+                )
+        return ret
+
+    async def async_resolve_media(self, item: MediaSourceItem) -> PlayMedia:
+        """Resolve media to a url."""
+        parts = item.identifier.split("/")
+        unique_id = parts[0]
+        cache_key = parts[2]
+        file_name = parts[3]
+        mime_type, _ = mimetypes.guess_type(parts[3])
+        assert isinstance(mime_type, str)
+        return PlayMedia(
+            f"/synology_dsm/{unique_id}/{cache_key}/{file_name}", mime_type
+        )
+
+    async def async_get_thumbnail(
+        self, cache_key: str, image_id: str, size: str, diskstation_unique_id: str
```

Comment:
```suggestion
        self, item: SynoPhotosItem, diskstation: SynologyDSMData
```

use the objects as is